packages/langium/src/default-module.ts

```diff
@@ -97,14 +97,25 @@ export function createDefaultModule(context: DefaultModuleContext): Module<Langi
  * Context required for creating the default shared dependency injection module.
  */
 export interface DefaultSharedModuleContext {
+    /**
+     * Represents an abstract language server connection
+     */
     connection?: Connection;
+    /**
+     * Factory function to create a {@link FileSystemProvider}.
+     *
+     * Langium exposes an `EmptyFileSystem` and `NodeFileSystem`, exported through `langium/node`.
+     * When running Langium as part of a vscode language server, using the `NodeFileSystem` is recommended,
+     * the `EmptyFileSystem` in every other use case.
```

Comment:
NodeFileSystem is also required in Node.js CLIs, e.g. to resolve cross-references to other files.